test-case/check-volume-levels.sh

```diff
@@ -0,0 +1,386 @@
+    dlogi "The test procedure is now complete. Killing the aplay process."
+    kill %1
+
+    # Measure, delete unnecessary wav files if passed
+    if measure_levels; then
+	rm -f "$APLAY_WAV" "$ARECORD_WAV1" "$ARECORD_WAV2" "$ARECORD_WAV3"
+    fi
+}
+
+#
+# Test #1 volume and mute switch controls
+#
+
+#  0...1s channels 1-4 max gain
+#  1...2s channels 1-4 different gains
+#  2...3s channels 1-4 nominal gain
+#  3...4s channels 1-4 muted
+#  4...5s channels 1-4 nominal gain
+#  5...7s channels 1-4 muted
+#  7...8s channels 1-4 max gain
+#  8...9s channels 1-4 muted gain
+#  9..10s channels 1-4 min gain
+# 10..11s channels 1-4 different gains
+run_test_1 () {
+    cset_max
+    cset_unmute
+    pause
+    arecord_helper "$ARECORD_WAV1" "$CAP_CHANNELS" 11 &
+    pause
+
+    cset_volume_diff1 "$CAP_CHANNELS"
+    pause
+
+    cset_nom
+    pause
+
+    cset_mute
+    pause
+
+    cset_unmute
+    pause
+
+    cset_mute
+    pause
+
+    cset_max
+    pause
+
+    cset_unmute
+    pause
+
+    cset_mutevol
+    pause
+
+    cset_min
+    pause
+
+    cset_volume_diff2 "$CAP_CHANNELS"
+
+    # Wait for arecord process to complete. %1 is aplay and %2 is the just launched arecord.
+    dlogi "Waiting"
+    wait %2
+    dlogi Ready.
+}
+
+#
+# Test #2, check gains preservation from previous to next
+#
+
+# 0..1s channels 1-4 previous gains
+# 1..2s channels 1-4 different mute switches
+# 2..3s channels 1-4 all muted
+run_test_2 () {
+    arecord_helper "$ARECORD_WAV2" "$CAP_CHANNELS" 4 &
+    pause
+
+    cset_nom
+    cset_mute_diff1 "$CAP_CHANNELS"
+    pause
+
+    cset_mute
+    pause
+
+    dlogi Waiting
+    wait %2
+    dlogi Ready.
+}
+
+#
+# Test #3, test mute switch preservation from previous to next
+#
+
+# 0..1s channels 1-4 muted
+# 1..2s channels 1-4  nominal gain
+# 2..3s different mute switches
+# 3..4s channels 1-4 nominal gain
+run_test_3 () {
+    arecord_helper "$ARECORD_WAV3" "$CAP_CHANNELS" 4 &
+    pause
+
+    cset_nom
+    pause
+
+    cset_mute_diff2 "$CAP_CHANNELS"
+    pause
+
+    cset_unmute
+
+    dlogi Waiting
+    wait %2
+    dlogi Ready.
+}
+
+#
+# Helper functions
+#
+
+cset_nom () {
+    dlogi Set nominal volume
+    amixer cset name="$CAP_VOLUME" $VOL_NOM
+}
+
+cset_max () {
+    dlogi Set maximum volume
+    amixer cset name="$CAP_VOLUME" $VOL_MAX
+}
+
+cset_min () {
+    dlogi Set minimum volume
+    amixer cset name="$CAP_VOLUME" $VOL_MIN
+}
+
+cset_mutevol () {
+    dlogi Set mute via volume
+    amixer cset name="$CAP_VOLUME" $VOL_MUTE
+}
+cset_mute () {
+    dlogi Set mute via switch
+    amixer cset name="$CAP_SWITCH" off
+}
+
+cset_unmute () {
+    dlogi Set unmute
+    amixer cset name="$CAP_SWITCH" on
+}
+
+pause () {
+    dlogi Pausing
+    sleep 1
+}
+
+cset_mute_diff1 () {
+    dlogi Set switch pattern 1
+    case "$1" in
+	4)
+	    amixer cset name="$CAP_SWITCH" off,on,on,off
+	    ;;
+	2)
+	    amixer cset name="$CAP_SWITCH" off,on
+	    ;;
+	1)
+	    amixer cset name="$CAP_SWITCH" off
+	    ;;
```

Comment:
Again that seems like a lot of whitespace and scrolling here, how about something a bit more compact?
```
4) amixer ... ;;
```

Reply:
OK